source/utilities.tex

```diff
@@ -6463,6 +6464,2043 @@
+
+\pnum
+\effects
+Direct-non-list-initializes \exposid{unex}
+with \tcode{std::forward<Args>(args)...}.
+
+\pnum
+\ensures
+\tcode{has_value()} is \tcode{false}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of \exposid{unex}.
+\end{itemdescr}
+
+\indexlibraryctor{expected}%
+\begin{itemdecl}
+template<class U, class... Args>
+    constexpr explicit expected(unexpect_t, initializer_list<U> il, Args&&... args);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\constraints
+\tcode{is_constructible_v<E, initializer_list<U>\&, Args...>} is \tcode{true}.
+
+\pnum
+\effects
+Direct-non-list-initializes \exposid{unex}
+with \tcode{il, std::forward<Args>(args)...}.
+
+\pnum
+\ensures
+\tcode{has_value()} is \tcode{false}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of \exposid{unex}.
+\end{itemdescr}
+
+\rSec3[expected.void.dtor]{Destructor}
+
+\indexlibrarydtor{expected}%
+\begin{itemdecl}
+constexpr ~expected();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+If \tcode{has_value()} is \tcode{false}, destroys \exposid{unex}.
+
+\pnum
+\remarks
+If \tcode{is_trivially_destructible_v<E>} is \tcode{true},
+then this destructor is a trivial destructor.
+\end{itemdescr}
+
+\rSec3[expected.void.assign]{Assignment}
+
+\indexlibrarymember{operator=}{expected}%
+\begin{itemdecl}
+constexpr expected& operator=(const expected& rhs);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+\begin{itemize}
+\item
+If \tcode{this->has_value() \&\& rhs.has_value()} is \tcode{true}, no effects.
+\item
+Otherwise, if \tcode{this->has_value()} is \tcode{true},
+equivalent to: \tcode{construct_at(addressof(\exposid{unex}), rhs.\exposid{unex}); \exposid{has_val} = false;}
+\item
+Otherwise, if \tcode{rhs.has_value()} is \tcode{true},
+destroys \exposid{unex} and sets \exposid{has_val} to true.
+\item
+Otherwise, equivalent to \tcode{\exposid{unex} = rhs.error()}.
+\end{itemize}
+
+\pnum
+\returns
+\tcode{*this}.
+
+\pnum
+\remarks
+This operator is defined as deleted unless
+\tcode{is_copy_assignable_v<E>} is \tcode{true} and
+\tcode{is_copy_constructible_v<E>} is \tcode{true}.
+\end{itemdescr}
+
+\indexlibrarymember{operator=}{expected}%
+\begin{itemdecl}
+constexpr expected& operator=(expected&& rhs) noexcept(@\seebelow@);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+\begin{itemize}
+\item
+If \tcode{this->has_value() \&\& rhs.has_value()} is \tcode{true}, no effects.
+\item
+Otherwise, if \tcode{this->has_value()} is \tcode{true}, equivalent to:
+\begin{codeblock}
+construct_at(addressof(@\exposid{unex}@), std::move(rhs.@\exposid{unex}@));
+@\exposid{has_val}@ = false;
+\end{codeblock}
+\item
+Otherwise, if \tcode{rhs.has_value()} is \tcode{true},
+destroys \exposid{unex} and sets \exposid{has_val} to \tcode{true}.
+\item
+Otherwise, equivalent to \tcode{\exposid{unex} = rhs.error()}.
+\end{itemize}
+
+\pnum
+\returns
+\tcode{*this}.
+
+\pnum
+\remarks
+The exception specification is
```

Comment:
If you need to make it a code block due to an overfull `\hbox`, remember to add the colon.
```suggestion
The exception specification is equivalent to
```

Reply:
No; see above.